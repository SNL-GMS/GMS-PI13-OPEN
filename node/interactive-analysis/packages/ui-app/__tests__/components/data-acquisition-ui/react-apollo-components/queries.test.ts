import * as graphql from 'react-apollo';
import {
  graphqlDefaultReferenceStationsQuery,
  graphqlTransferredFilesByTimeRangeQueryQuery
} from '../../../../src/ts/components/data-acquisition-ui/react-apollo-components/queries';
// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
// const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

describe('queries ', () => {
  const graphqlold = graphql.graphql;
  beforeAll(() => {
    (graphql as any).graphql = jest.fn() as any;
  });
  afterAll(() => {
    (graphql as any).graphql = graphqlold;
  });
  it('should be defined', () => {
    expect(graphqlTransferredFilesByTimeRangeQueryQuery).toBeDefined();
    expect(graphqlDefaultReferenceStationsQuery).toBeDefined();
  });
  it('should have graphqlTransferredFilesByTimeRangeQueryQuery', () => {
    graphqlTransferredFilesByTimeRangeQueryQuery();
    expect(graphql.graphql).toBeCalledTimes(1);
    expect((graphql.graphql as any).mock.calls[0]).toMatchSnapshot();
  });
  it('should have graphqlTransferredFilesByTimeRangeQueryQuery', () => {
    graphqlDefaultReferenceStationsQuery();
    expect(graphql.graphql).toBeCalledTimes(2);
    expect((graphql.graphql as any).mock.calls[1]).toMatchSnapshot();
  });
});
