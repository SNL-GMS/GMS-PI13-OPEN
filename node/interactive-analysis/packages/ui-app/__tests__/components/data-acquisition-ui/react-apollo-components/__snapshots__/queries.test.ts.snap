// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries  should have graphqlTransferredFilesByTimeRangeQueryQuery 1`] = `
Array [
  Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "kind": "OperationDefinition",
        "name": Object {
          "kind": "Name",
          "value": "transferredFilesByTimeRange",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [
                Object {
                  "kind": "Argument",
                  "name": Object {
                    "kind": "Name",
                    "value": "timeRange",
                  },
                  "value": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "timeRange",
                    },
                  },
                },
              ],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "transferredFilesByTimeRange",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "directives": Array [],
                    "kind": "FragmentSpread",
                    "name": Object {
                      "kind": "Name",
                      "value": "FileGapsFragment",
                    },
                  },
                ],
              },
            },
          ],
        },
        "variableDefinitions": Array [
          Object {
            "defaultValue": undefined,
            "directives": Array [],
            "kind": "VariableDefinition",
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "TimeRange",
                },
              },
            },
            "variable": Object {
              "kind": "Variable",
              "name": Object {
                "kind": "Name",
                "value": "timeRange",
              },
            },
          },
        ],
      },
      Object {
        "directives": Array [],
        "kind": "FragmentDefinition",
        "name": Object {
          "kind": "Name",
          "value": "FileGapsFragment",
        },
        "selectionSet": Object {
          "kind": "SelectionSet",
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "stationName",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "channelNames",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "startTime",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "endTime",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "duration",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "location",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "priority",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "typeCondition": Object {
          "kind": "NamedType",
          "name": Object {
            "kind": "Name",
            "value": "FileGap",
          },
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 303,
      "start": 0,
    },
  },
  Object {
    "name": "transferredFilesByTimeRangeQuery",
    "options": [Function],
  },
]
`;

exports[`queries  should have graphqlTransferredFilesByTimeRangeQueryQuery 2`] = `
Array [
  Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "kind": "OperationDefinition",
        "name": Object {
          "kind": "Name",
          "value": "defaultReferenceStations",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "defaultReferenceStations",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "directives": Array [],
                    "kind": "FragmentSpread",
                    "name": Object {
                      "kind": "Name",
                      "value": "ReferenceStationFragment",
                    },
                  },
                ],
              },
            },
          ],
        },
        "variableDefinitions": Array [],
      },
      Object {
        "directives": Array [],
        "kind": "FragmentDefinition",
        "name": Object {
          "kind": "Name",
          "value": "ReferenceStationFragment",
        },
        "selectionSet": Object {
          "kind": "SelectionSet",
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "id",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "name",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "latitude",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "longitude",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "elevation",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "stationType",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "description",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "defaultChannel",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "directives": Array [],
                    "kind": "FragmentSpread",
                    "name": Object {
                      "kind": "Name",
                      "value": "ReferenceChannelFragment",
                    },
                  },
                ],
              },
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "networks",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "id",
                    },
                    "selectionSet": undefined,
                  },
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "name",
                    },
                    "selectionSet": undefined,
                  },
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "monitoringOrganization",
                    },
                    "selectionSet": undefined,
                  },
                ],
              },
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "location",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "directives": Array [],
                    "kind": "FragmentSpread",
                    "name": Object {
                      "kind": "Name",
                      "value": "LocationFragment",
                    },
                  },
                ],
              },
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "sites",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "directives": Array [],
                    "kind": "FragmentSpread",
                    "name": Object {
                      "kind": "Name",
                      "value": "ReferenceSiteFragment",
                    },
                  },
                ],
              },
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "dataAcquisition",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "dataAcquisition",
                    },
                    "selectionSet": undefined,
                  },
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "interactiveProcessing",
                    },
                    "selectionSet": undefined,
                  },
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "automaticProcessing",
                    },
                    "selectionSet": undefined,
                  },
                ],
              },
            },
          ],
        },
        "typeCondition": Object {
          "kind": "NamedType",
          "name": Object {
            "kind": "Name",
            "value": "ReferenceStation",
          },
        },
      },
      Object {
        "directives": Array [],
        "kind": "FragmentDefinition",
        "name": Object {
          "kind": "Name",
          "value": "ReferenceSiteFragment",
        },
        "selectionSet": Object {
          "kind": "SelectionSet",
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "id",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "name",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "channels",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "directives": Array [],
                    "kind": "FragmentSpread",
                    "name": Object {
                      "kind": "Name",
                      "value": "ReferenceChannelFragment",
                    },
                  },
                ],
              },
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "location",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "directives": Array [],
                    "kind": "FragmentSpread",
                    "name": Object {
                      "kind": "Name",
                      "value": "LocationFragment",
                    },
                  },
                ],
              },
            },
          ],
        },
        "typeCondition": Object {
          "kind": "NamedType",
          "name": Object {
            "kind": "Name",
            "value": "ReferenceSite",
          },
        },
      },
      Object {
        "directives": Array [],
        "kind": "FragmentDefinition",
        "name": Object {
          "kind": "Name",
          "value": "LocationFragment",
        },
        "selectionSet": Object {
          "kind": "SelectionSet",
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "latitudeDegrees",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "longitudeDegrees",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "elevationKm",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "typeCondition": Object {
          "kind": "NamedType",
          "name": Object {
            "kind": "Name",
            "value": "Location",
          },
        },
      },
      Object {
        "directives": Array [],
        "kind": "FragmentDefinition",
        "name": Object {
          "kind": "Name",
          "value": "ReferenceChannelFragment",
        },
        "selectionSet": Object {
          "kind": "SelectionSet",
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "id",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "name",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "channelType",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "sampleRate",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "position",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "northDisplacementKm",
                    },
                    "selectionSet": undefined,
                  },
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "eastDisplacementKm",
                    },
                    "selectionSet": undefined,
                  },
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "verticalDisplacementKm",
                    },
                    "selectionSet": undefined,
                  },
                ],
              },
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "actualTime",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "systemTime",
              },
              "selectionSet": undefined,
            },
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "depth",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "typeCondition": Object {
          "kind": "NamedType",
          "name": Object {
            "kind": "Name",
            "value": "ReferenceChannel",
          },
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 1497,
      "start": 0,
    },
  },
  Object {
    "name": "defaultStationsQuery",
  },
]
`;
